src/main/java/org/generationcp/middleware/dao/UserRoleDao.java

```diff
@@ -0,0 +1,132 @@
+package org.generationcp.middleware.dao;
+
+import org.generationcp.middleware.domain.workbench.RoleType;
+import org.generationcp.middleware.exceptions.MiddlewareQueryException;
+import org.generationcp.middleware.pojos.workbench.CropType;
+import org.generationcp.middleware.pojos.workbench.PermissionsEnum;
+import org.generationcp.middleware.pojos.workbench.UserRole;
+import org.generationcp.middleware.pojos.workbench.WorkbenchUser;
+import org.hibernate.Criteria;
+import org.hibernate.HibernateException;
+import org.hibernate.SQLQuery;
+import org.hibernate.criterion.Projections;
+import org.hibernate.criterion.Restrictions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class UserRoleDao extends GenericDAO<UserRole, Long> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(UserRole.class);
+
+		private static final String HAS_INSTANCE_ROLE_WITH_ADD_PROGRAM_PERMISSION_SQL = "select ur.role_id from users_roles ur " //
+			+"  inner join role r on ur.role_id = r.id " //
+			+"  inner join role_type type on type.role_type_id = r.role_type_id " //
+			+"  inner join role_permission rp on r.id = rp.role_id " //
+			+"  inner join permission p on rp.permission_id = p.permission_id " //
+			+"  where (r.role_type_id = " + RoleType.INSTANCE.getId() + " and p.name in ('" + PermissionsEnum.ADMIN.toString() + "', '"+ PermissionsEnum.CROP_MANAGEMENT.toString() //
+			+"','"+ PermissionsEnum.ADD_PROGRAM.toString()+"', '"+ PermissionsEnum.MANAGE_PROGRAMS+"'))" //
+			+"  and ur.userid = :userId and r.active = 1";//
+
+
+		private static final String GET_CROPS_WITH_ADD_PROGRAM_PERMISSION_FOR_A_CROP_ROLE_SQL = "select distinct ur.crop_name from users_roles ur "//
+		+"  inner join role r on ur.role_id = r.id " //
+		+"  inner join role_type type on type.role_type_id = r.role_type_id " //
+		+"  inner join role_permission rp on r.id = rp.role_id " //
+		+"  inner join permission p on rp.permission_id = p.permission_id "//
+		+"  where (r.role_type_id = " + RoleType.CROP.getId() + " and p.name in ('" + PermissionsEnum.CROP_MANAGEMENT.toString() + "', '"+ PermissionsEnum.MANAGE_PROGRAMS.toString() //
```

Comment:
Same here, use `query.setParameterList `instead of concatenating the different permissions to the query string. Create constant for the list of permissions to be checked